src/tracer.cc

```diff
@@ -337,13 +342,14 @@ TraceManager::Trace::CaptureTimestamp(
           << "{\"name\":\"" << name << "\",\"ns\":" << timestamp_ns << "}]}";
     } else if (setting_->mode_ == TRACE_MODE_OPENTELEMETRY) {
 #ifndef _WIN32
-      otel_common::SystemTimestamp otel_timestamp{
-          (time_offset_ + std::chrono::nanoseconds{timestamp_ns})};
-      if (trace_span_ == nullptr) {
-        InitSpan(otel_timestamp);
+      if (otel_context_.HasKey(kRootSpan)) {
```

Comment:
constructed `Trace` object may not have root span?

Reply:
No, it should have root span, since it was added during the initialization. I will remove these extra checks